### PR TITLE
samples: openthread: adjust logging options to logger v2

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
With the last zephyr upmerge the default logger is LOG2 which does not use strdup function. Any related Kconfig options stops to be applicable.
Adjust Thread samples configuration to the changes.

### CODE REVIEW

File: doc/nrf/ug_thread_configuring.rst [hunk 1]
```diff
@@ -190,13 +190,10 @@ Select one of the following logging levels to customize the logging output:
...
-The more detailed logging level you select, the more logging buffers you need to be able to see all messages.
-The buffer size must also be increased.
-Use the following Kconfig options for this purpose:
-
-* :kconfig:option:`CONFIG_LOG_STRDUP_BUF_COUNT` - This option specifies the number of logging buffers.
-* :kconfig:option:`CONFIG_LOG_STRDUP_MAX_STRING` - This option specifies the size of logging buffers.
+The more detailed logging level you select, the bigger logging buffer you need to have to see all messages.
+Use the following Kconfig option for this purpose:
 
+* :kconfig:option:`CONFIG_LOG_BUFFER_SIZE` - This option specifies the number of logging buffers.
```
Comment (edmont) [inline doc/nrf/ug_thread_configuring.rst:196]: It seems it's not exactly the number of buffers: _Number of bytes dedicated for the logger internal buffer_

File: samples/openthread/cli/overlay-logging.conf [hunk 2]
```diff
@@ -25,4 +25,5 @@ CONFIG_OPENTHREAD_DEBUG=y
 # Option for configuring log level in OpenThread
 CONFIG_OPENTHREAD_LOG_LEVEL_INFO=y
 
-# Adjust log strdup settings
+# Increase log buffer size for OpenThread
+CONFIG_LOG_BUFFER_SIZE=4048
```
Comment (edmont) [inline samples/openthread/cli/overlay-logging.conf:29]: Maybe move this configuration to `nrf/subsys/net/openthread/Kconfig.defconfig` instead?

```
config LOG_BUFFER_SIZE
	default 4048 if LOG
```

**Review by edmont — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

File: doc/nrf/ug_thread_configuring.rst [hunk 3]
```diff
@@ -190,13 +190,10 @@ Select one of the following logging levels to customize the logging output:
...
-The more detailed logging level you select, the more logging buffers you need to be able to see all messages.
-The buffer size must also be increased.
-Use the following Kconfig options for this purpose:
-
-* :kconfig:option:`CONFIG_LOG_STRDUP_BUF_COUNT` - This option specifies the number of logging buffers.
-* :kconfig:option:`CONFIG_LOG_STRDUP_MAX_STRING` - This option specifies the size of logging buffers.
+The more detailed logging level you select, the bigger logging buffer you need to have to see all messages.
+Use the following Kconfig option for this purpose:
 
+* :kconfig:option:`CONFIG_LOG_BUFFER_SIZE` - This option specifies the number of bytes dedicated for the logger internal buffer.
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_configuring.rst:196]:
```suggestion
* :kconfig:option:`CONFIG_LOG_BUFFER_SIZE` - This option specifies the number of bytes dedicated to the logger internal buffer.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
Very small nitpick from me, otherwise very good.

**Review by wiba-nordic — APPROVED** [review]

**Pull request merged**